flow/util/misc.py

```diff
@@ -393,6 +394,28 @@ def parallel_executor(func, iterable, **kwargs):
     return parallel_executor
 
 
+def _deprecated_warning(
+    deprecation: str,
+    alternative: str,
+    deprecated_in: str,
+    removed_in: str,
+    category: "Warning" = FutureWarning,
+):
+    warnings.warn(
+        " ".join(
+            (
+                deprecation,
+                "has been deprecated as of",
+                deprecated_in,
+                "and will be removed in",
+                removed_in + ".",
+                alternative,
+            )
+        ),
+        category,
```

Comment:
Issue the warning so that it appears to come from the caller and not from this function. See https://docs.python.org/3/library/warnings.html#warnings.warn
```suggestion
        category,
        stacklevel=2,
```

Reply:
Good catch!